x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -144,7 +146,8 @@ protected static Batch<LogicalPlan> operators() {
             new PushDownEnrich(),
             new PushDownAndCombineOrderBy(),
             new PruneOrderByBeforeStats(),
-            new PruneRedundantSortClauses()
+            new PruneRedundantSortClauses(),
+            new PartiallyFoldCase()
```

Comment:
It doesn't make much of a difference however I move the rule after `ConstantFolding` so the reduction rules are grouped together.

Reply:
++